test/cypress/tests/dragnDrop.cy.ts

```diff
@@ -0,0 +1,368 @@
+      });
+  });
+
+  it('should have block dragover style on the bottom of target block', function () {
+    cy.get('[data-cy=editorjs]')
+      .get('div.ce-block')
+      .first()
+      .click();
+
+    const dataTransfer = new DataTransfer();
+
+    // eslint-disable-next-line cypress/require-data-selectors
+    cy.get('.ce-toolbar__settings-btn')
+      .trigger('dragstart', { dataTransfer });
+
+    cy.get('[data-cy=editorjs]')
+      .get('div.ce-block')
+      .eq(2)
+      .trigger('dragenter')
+      .then((blocks) => {
+        // Get the target block rect.
+        const targetBlockRect = blocks[0].getBoundingClientRect();
+        const yShiftFromMiddleLine = 20;
+        // Dragover on target block little bit below the middle line.
+        const dragOverYCoord =
+          targetBlockRect.y +
+          (targetBlockRect.height / 2 + yShiftFromMiddleLine);
+
+        cy.get('[data-cy=editorjs]')
+          .get('div.ce-block')
+          .eq(2)
+          .trigger('dragover', {
+            clientX: targetBlockRect.x,
+            clientY: dragOverYCoord,
+          })
+          .then(($element) => {
+            // check for dragover top style on target element.
+            const classes = $element.attr('class').split(' ');
+
+            expect(classes).to.include('ce-block--drop-target');
+            expect(classes).to.include('ce-block--drop-target-bottom');
+          });
+      });
+  });
+
+  it('should drag the first block and drop after the last block.', function () {
+    cy.get('[data-cy=editorjs]')
+      .get('div.ce-block')
+      .first()
+      .click();
+
+    const dataTransfer = new DataTransfer();
+
+    // eslint-disable-next-line cypress/require-data-selectors
+    cy.get('.ce-toolbar__settings-btn')
+      .trigger('dragstart', { dataTransfer });
+
+    cy.get('[data-cy=editorjs]')
+      .get('div.ce-block')
+      .eq(2)
+      .trigger('dragenter')
+      .then((blocks) => {
+        // Get the target block rect.
+        const targetBlockRect = blocks[0].getBoundingClientRect();
+        const yShiftFromMiddleLine = 20;
+        // Dragover on target block little bit below the middle line.
+        const dragOverYCoord =
+          targetBlockRect.y +
+          (targetBlockRect.height / 2 + yShiftFromMiddleLine);
+
+        cy.get('[data-cy=editorjs]')
+          .get('div.ce-block')
+          .eq(2)
+          .trigger('dragover', {
+            clientX: targetBlockRect.x,
+            clientY: dragOverYCoord,
+          })
+          .trigger('drop', { dataTransfer });
+      });
+
+    // eslint-disable-next-line cypress/require-data-selectors
+    cy.get('.ce-toolbar__settings-btn')
+      .trigger('dragend', { dataTransfer });
+
+    cy.get('[data-cy=editorjs]')
+      .get('div.ce-block')
+      .then((blocks) => {
+        expect(blocks[0].textContent).to.eq('Block 1');
+        expect(blocks[1].textContent).to.eq('Block 2');
+        expect(blocks[2].textContent).to.eq('Block 0');
```

Comment:
Also would be great to test output data from the .save method.

Reply:
will update it.